chrome-launcher/chrome-launcher.ts

```diff
@@ -19,46 +19,75 @@
 
 import * as childProcess from 'child_process';
 import * as fs from 'fs';
-import * as path from 'path';
 import * as chromeFinder from './chrome-finder';
-import {ask} from './ask';
 import {DEFAULT_FLAGS} from './flags';
-
-const mkdirp = require('mkdirp');
+import {makeTmpDir, defaults, delay} from './utils';
 import * as net from 'net';
 const rimraf = require('rimraf');
 const log = require('../lighthouse-core/lib/log');
 const spawn = childProcess.spawn;
 const execSync = childProcess.execSync;
 const isWindows = process.platform === 'win32';
+const _SIGINT = 'SIGINT';
+const _SIGINT_EXIT_CODE = 130;
+const _SUPPORTED_PLATFORMS = ['darwin', 'linux', 'win32'];
+
+type SupportedPlatforms = 'darwin'|'linux'|'win32';
+
+export interface Options {
+  startingUrl?: string;
+  chromeFlags?: Array<string>;
+  port?: number;
+  handleSIGINT?: boolean;
+  chromePath?: string;
+}
+
+export interface LaunchedChrome {
+  pid: number;
+  port: number;
+  kill: () => Promise<{}>;
+}
+
+export async function launch(opts: Options = {}): Promise<LaunchedChrome> {
+  opts.handleSIGINT = defaults(opts.handleSIGINT, true);
 
-export class ChromeLauncher {
-  prepared = false;
+  const instance = new ChromeLauncher(opts);
+  await instance.launch();
+
+  // Kill spawned Chrome process in case of ctrl-C.
+  if (opts.handleSIGINT) {
+    process.on(_SIGINT, async () => {
+      await instance.kill();
+      process.exit(_SIGINT_EXIT_CODE);
+    });
+  }
+
+  return {pid: instance.pid!, port: instance.port, kill: instance.kill};
+}
+
+class ChromeLauncher {
```

Comment:
as mentioned earlier, i think we should rename this class, something that doesn't sound so public.


how about just `Launcher` or maybe a `LauncherImpl` ?